lib/src/utils.dart

```diff
@@ -31,19 +31,45 @@ final Logger logger = Logger('dependency_validator');
 String bulletItems(Iterable<String> items) =>
     items.map((l) => '  * $l').join('\n');
 
-/// Returns the name of the package referenced in the `include:` directive in an
-/// analysis_options.yaml file, or null if there is not one.
-String? getAnalysisOptionsIncludePackage({String? path}) {
+/// Returns the names of the packages referenced in the `include:` directive in an
+/// analysis_options.yaml file, or null if there are none.
+///
+/// Supports both single string includes:
+///   include: package:flutter_lints/flutter.yaml
+///
+/// And list includes:
+///   include:
+///     - package:flutter_lints/flutter.yaml
+///     - package:another_package/config.yaml
+Set<String>? getAnalysisOptionsIncludePackage({String? path}) {
   final optionsFile = File(p.join(path ?? p.current, 'analysis_options.yaml'));
   if (!optionsFile.existsSync()) return null;
 
   final YamlMap? analysisOptions = loadYaml(optionsFile.readAsStringSync());
   if (analysisOptions == null) return null;
 
-  final String? include = analysisOptions['include'];
-  if (include == null || !include.startsWith('package:')) return null;
+  final dynamic include = analysisOptions['include'];
+  if (include == null) return null;
+
+  final Set<String> packageNames = <String>{};
```

Comment:
nit: We generally follow this rule https://dart.dev/effective-dart/design#dont-redundantly-type-annotate-initialized-local-variables so the explicit declaration isn't necessary.